### ISSUE TITLE
15.0.1 (24A348)The software is often automatically closed. [bug]

### ISSUE THREAD

**Issue by 1052971551** [noah-nuebling/Mac-Mouse-Fix#1147]
## Description

After running in 15.0.1 (24A348) for a period of time, the software will always be automatically turned off. You need to manually open the software and click to enable Mac Mouse Fix to continue using it. This problem has been bothering me for a long time. The problem of uninstalling and reinstalling the software still exists.

## Steps to reproduce

1. Start Mac Mouse Fix for normal use

2. After a period of working time, the hairstyle roller suddenly got stuck. When I opened Mac Mouse Fix, I found that the enable option was changed to off.

3. It can only be used after restarting. The problem is that it will automatically turn off after a period of time.

## Context

Mac Mouse Fix Version: 3.0.3
macOS Version: 15.0.1 (24A348)
Mouse Model: –

## Additional Comments

–

---

### Console Logs

–

### Crash Reports

–

**Comment by noah-nuebling** [comment]
Hi @1052971551! Thank you for the report.

I'm not sure what might cause this issue. To further investigate this, the next time you find that the app has been automatically turned off:

1. Run the following command in the Terminal, and let me know what the output is:
    ```
    launchctl print gui/501/com.nuebling.mac-mouse-fix.helper
    ```
2. Open `Console.app`, go to `Crash Reports` and see if you can find any recent reports related to Mac Mouse Fix. If so, please send the newest report to me. 

This information would give us more clues as to whether the problem might be related to crashes inside Mac Mouse Fix, or related to problems inside Apple's system for running processes in the background. \
(This system can be interacted with using the `launchctl` command-line-tool)

Thank you, and I'm sorry about the issues. 

If you are interested in a refund [click here](https://redirect.macmousefix.com/?target=mmf-apply-for-refund). And if you have any more questions, just let me know!

**Comment by 1052971551** [comment]
> Run the following command in the Terminal, and let me know what the output is:
> launchctl print gui/501/com.nuebling.mac-mouse-fix.helper

gui/501/com.nuebling.mac-mouse-fix.helper = {
	active count = 5
	path = (submitted by smd.88)
	type = Submitted
	managed_by = com.apple.xpc.ServiceManagement
	state = running

	program identifier = Contents/Library/LoginItems/Mac Mouse Fix Helper.app/Contents/MacOS/Mac Mouse Fix Helper (mode: 2)
	parent bundle identifier = com.nuebling.mac-mouse-fix
	parent bundle version = 22854
	BTM uuid = BF46A8A7-A44F-4B8B-B725-6F89593153EB
	stdout path = /dev/null
	stderr path = /dev/null
	inherited environment = {
		SSH_AUTH_SOCK => /private/tmp/com.apple.launchd.BWTxVe8e6i/Listeners
	}

	default environment = {
		PATH => /usr/bin:/bin:/usr/sbin:/sbin
	}

	environment = {
		XPC_SERVICE_NAME => com.nuebling.mac-mouse-fix.helper
	}

	domain = gui/501 [100003]
	asid = 100003
	minimum runtime = 10
	exit timeout = 5
	nice = -10
	runs = 1
	pid = 608
	immediate reason = speculative
	forks = 0
	execs = 1
	initialized = 1
	trampolined = 1
	started suspended = 0
	proxy started suspended = 0
	last exit code = (never exited)

	dynamic endpoints = {
		"com.nuebling.mac-mouse-fix.helper" = {
			port = 0xa27d3
			active = 1
			managed = 0
			reset = 0
			hide = 0
			watching = 0
		}
	}

	pid-local endpoints = {
		"com.apple.tsm.portname" = {
			port = 0x8d803
			active = 1
			managed = 0
			reset = 0
			hide = 0
			watching = 0
		}
		"com.apple.axserver" = {
			port = 0x9ab03
			active = 1
			managed = 0
			reset = 0
			hide = 0
			watching = 0
		}
		"com.apple.coredrag" = {
			port = 0x8bc03
			active = 1
			managed = 0
			reset = 0
			hide = 0
			watching = 0
		}
	}

	spawn type = interactive (4)
	jetsam priority = 40
	jetsam memory limit (active) = (unlimited)
	jetsam memory limit (inactive) = (unlimited)
	jetsamproperties category = daemon
	submitted job. ignore execute allowed
	jetsam thread limit = 32
	cpumon = default
	job state = running
	probabilistic guard malloc policy = {
		activation rate = 1/1000
		sample rate = 1/0
	}

	properties = partial import | keepalive | runatload | resolve program | has LWCR
}


> 
> Open Console.app, go to Crash Reports and see if you can find any recent reports related to Mac Mouse Fix. If so, please send the newest report to me.


￼
This is the only crash report. I'm not sure if it's this one.

**Comment by noah-nuebling** [comment]
Thank you very much for the data!

From a quick check of the `launchctl` output, it looks like the macOS run-stuff-in-the-background-system is working normally for Mac Mouse Fix. Can you see a process called "Mac Mouse Fix Helper" in Activity Monitor when the issue occurs? That would confirm that there really no problem in the `launchctl` system.

> This is the only crash report. I'm not sure if it's this one.

I can't see a crash report attached to your message. Did you perhaps forget to attach it, or did you mean the `launchcl` output?

**Comment by noah-nuebling** [comment]
Also, I just thought of two more questions that might help diagnose the problem: 

- Does the scrolling stop working entirely or is it just not smooth anymore?
- Do you use the Mac Mouse Fix to assign actions or gestures to your mouse buttons, or do you use it just for scrolling? How do the mouse buttons behave when MMF stops working?

**Comment by 1052971551** [comment]
At present, there is no problem with the software, and this problem will occur from time to time. When this problem occurs, turn on the software again and the switch is turned off.

I guess the service has been terminated. Every time I turn on the software again, this switch can be used normally.

**Comment by noah-nuebling** [comment]
Thank you for your message!

> I guess the service has been terminated.

That is plausible! But what's weird is that the `launchctl` output, which you sent me, suggests that Mac Mouse Fix was still enabled and running normally from the perspective of macOS. But since your experience was that the app didn't work anymore and that the "Enable Mac Mouse Fix" switch is turned off when you looked inside the app - something else must be going wrong - and I don't know what.

Since I cannot reproduce this issue and have never experienced it myself, I can unfortunately not investigate the bug myself.

That's why - **once you experience the problem again**, it would be very helpful if you try to obtain the following info:

1. Open the "Activity Monitor.app" application and check if there is a process running called "Mac Mouse Fix Helper".
2. Open "Console.app", then go to "Crash Reports" and see  if you can find any reports related to Mac Mouse Fix. If so, send them to me.
3. If you want, you could also monitor the "logs" that Mac Mouse Fix writes to the console. If you can catch the logs that Mac Mouse Fix writes while the bug happens, that would be a great way to diagnose the problem by letting us inspect the internal state of the app. 
    - However, capturing this information would require you to always have Console.app running in the background, recording logs so that, when the bug happens again, you will have recorded all the information about how it happened. If you're interested in this, please let me know and I will send you instructions (alongside a debug build of Mac Mouse Fix which logs more detailed information)

It would also be interesting to hear more details of what happens when you experience the bug: Does your mouse behave *exactly* as if Mac Mouse Fix was turned off (meaning that your scrolling and your mouse buttons aren't affected at all) or is there some non-standard behavior you notice (For example: You cannot scroll inside any windows at all anymore)

Also: Are you using only the scrolling features of Mac Mouse Fix or are you also using the button features? I ask because multiple people had similar issues but only with scrolling. So perhaps these issues are related.

Thank you very much for helping to make Mac Mouse Fix better.

---

我在 Claude AI 的帮助下将此消息翻译成中文，供您参考：

感谢您的留言！

> 我猜服务已经终止了。

这种情况确实有可能！但奇怪的是，您发送的 `launchctl` 输出显示，从 macOS 的角度来看，Mac Mouse Fix 仍然处于启用和正常运行状态。但既然您发现应用程序不再工作，而且在应用程序内部看到"启用 Mac Mouse Fix"开关处于关闭状态 - 那么一定是其他地方出了问题 - 但我不知道具体是什么问题。

由于我无法重现这个问题，而且我自己也从未遇到过，所以很遗憾我无法亲自调查这个故障。

因此 - **当您再次遇到这个问题时**，如果您能够获取以下信息将会非常有帮助：

1. 打开"活动监视器.app"应用程序，检查是否有名为"Mac Mouse Fix Helper"的进程在运行。
2. 打开"控制台.app"，然后转到"崩溃报告"，看看是否能找到与 Mac Mouse Fix 相关的报告。如果找到，请发送给我。
3. 如果您愿意，您也可以监控 Mac Mouse Fix 写入控制台的"日志"。如果您能捕获到故障发生时 Mac Mouse Fix 写入的日志，这将是一个很好的诊断方法，让我们可以检查应用程序的内部状态。
    - 不过，捕获这些信息需要您始终在后台运行控制台应用程序记录日志，这样当故障再次发生时，您就会记录下关于故障发生过程的所有信息。如果您对此感兴趣，请告诉我，我会发送给您相关说明（以及一个能记录更详细信息的 Mac Mouse Fix 调试版本）

另外听听您遇到故障时的具体情况也会很有帮助：您的鼠标表现是否*完全*像 Mac Mouse Fix 被关闭时一样（这意味着您的滚动和鼠标按钮完全不受影响），还是您注意到了一些非标准的行为（例如：您完全无法在任何窗口内滚动）

还有：您是只使用 Mac Mouse Fix 的滚动功能，还是也使用按钮功能？我这样问是因为有多个用户遇到类似的问题，但只出现在滚动功能上。所以这些问题可能是相关的。

非常感谢您帮助改进 Mac Mouse Fix。

**Comment by 1052971551** [comment]
Hello, this problem appeared again after a day and a half, but a process called Mac mouse fix helper can be found in the activity monitor, but there is no crash report in the console. Now the mouse scrolling is obviously stuck, and it is not smooth when mouse fix is turned on.

**Comment by noah-nuebling** [comment]
Hello @1052971551, thank you for the feedback, I would like to address this, but unfortunately I can't investigate the bug without more detailed diagnostic information. Most importantly Console logs.

I understand that the process of always having the Console.app running in the background to catch the bug and then noting down the time and copy-pasting all the logs is not a great experience. Yesterday, I made some plans on how to make this experience easier for the future, but that will take a while to add into the app, and unfortunately I think, right now, running Console.app in the background is the easiest way to collect this diagnostic information.

There's also an alternative way to collect the diagnostics, where you would either install a specific "configuration profile" in System Settings or run a few terminal commands to enable recording Console Logs in the background - then you would *not* have to run Console.app all the time to record the diagnostics. If you'd be more comfortable to do one of those alternative approaches to gather diagnostics, then let me know and I can send you full instructions how to do it.

So in summary: I'd like to investigate your bug further, but I'd need more information for that which I unfortunately cannot gather by myself. But I totally understand if you have other things to do. 

I hope this makes sense and clears things up. If you have any more information about the issue, let me know!

**Comment by byalextran** [comment]
i'm not sure if this is related, but since i discovered and started using mouse fix over the weekend, there have been at least a couple times when scrolling stops working. up until this point, i had rebooted the computer to fix it. but just made the connection that it could be a mouse fix issue, so when this happened again just now, i launched mouse fix and disabled it (via the toggle) and then re-enabled it and the scrolling started working again. 

i'll keep an eye on it and leave console.app open to see if i can capture anything the next time it happens again.

**Comment by noah-nuebling** [comment]
Great, thank you @byalextran! That sure sounds like it's related.

If you decide to go ahead with capturing Console logs, here are some more *tips to make the logs as useful as possible*:
1. Use [this debug build](https://github.com/user-attachments/files/17517713/Mac.Mouse.Fix.app.zip) of Mac Mouse Fix, it will log much more information to the Console.
2. To log the additional information that the debug build outputs, include  'Info' and 'Debug' messages under `Console.app > Action`:
    <img src="https://github.com/user-attachments/assets/cca9ce66-2525-4add-8d7c-03adf3ec1232" data-canonical-src="https://gyazo.com/eb5c5741b6a9a16c692170a41a49c858.png" width="500" />
3. Once the bug happens, make a screenshot (Command-Shift-3)
    (The screenshot's timestamp will help me find the issue in the logs.)
4. You can select all messages in Console.app with Command-A - that way you can easily copy-paste them.
5. Once you're done, don't forget to send:
    1. Console logs
    2. Screenshot (Or alternatively the time when the bug appeared e.g. "It happened between 10:00 - 10:30")
    
On privacy: Private information is normally redacted from Console logs, but there will be thousands or millions of logs from different processes on the system, and I'm not sure whether private information could be included in one of them by the developers by mistake. If you're concerned about this, you might prefer sending the logs to [my email](https://redirect.macmousefix.com/?target=mail-to-noah) instead of posting them here publicly.

Let me know if you have any more questions. If this is too much stuff, and you have other things to do, I also totally understand of course. Thank you for offering your help.

**Comment by byalextran** [comment]
great, got the debug build installed and will wait for this to happen again and submit the logs.
